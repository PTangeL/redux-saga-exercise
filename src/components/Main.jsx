import React from 'react'
import TinderContainer from '../containers/TinderContainer'
import Header from './Header'
import Menu from './Navigations/Menu'

class Main extends React.Component {
  constructor () {
    super()
    this.toggleMenu = this.toggleMenu.bind(this)
    this.state = {
      menu: { open: true }
    }
  }

  toggleMenu () {
    this.setState({ menu: { open: !this.state.menu.open } })
  }

  render () {
    return (
      <div id="page-wrap">
        <Menu
          pageWrapId="page-wrap"
          isOpen={this.state.menu.open}
          toggleMenu={this.toggleMenu}
        />

        <nav className="navbar navbar-default navbar-fixed-top navbar-custom">
          <div className="container">
            <div className="navbar-header">
              <a className="navbar-brand" href="https://leanjs.com" target="_blank" rel="noopener noreferrer">LeanJS</a>
            </div>
            <ul className="nav navbar-nav pull-right">
              <li>
                <button onClick={this.toggleMenu} className="btn btn-lg btn-outline">
                  <i className="fa fa-heart" />
                </button>
              </li>
            </ul>
          </div>
        </nav>

        <Header />

        <section id="beers">
          <div className="container">
            <TinderContainer />
          </div>
        </section>

        <footer className="text-center">
          <div className="footer-above">
            <div className="container">
              <div className="row">
                <div className="footer-col col-md-4">
                  <h3>Main Location</h3>
                  <p>
                    <span>1 Fore Stree</span>
                    <br />
                    <span>London, EC2Y 5EJ</span>
                    <br />
                    <span className="fa fa-map-marker" />
                    <a target="_blank" href="https://leanjs.com/training/react-redux-graphql-london" rel="noopener noreferrer"> Other Locations </a>
                  </p>
                </div>
                <div className="footer-col col-md-4">
                  <h3>Around the Web</h3>
                  <ul className="list-inline">
                    <li>
                      <a target="_blank" href="https://github.com/leanjscom" className="btn-social btn-outline" rel="noopener noreferrer"><i className="fa fa-fw fa-github" /></a>
                    </li>
                    <li>
                      <a target="_blank" href="https://twitter.com/leanjscom" className="btn-social btn-outline" rel="noopener noreferrer"><i className="fa fa-fw fa-twitter" /></a>
                    </li>
                    <li>
                      <a target="_blank" href="https://www.instagram.com/leanjscom/" className="btn-social btn-outline" rel="noopener noreferrer"><i className="fa fa-fw fa-instagram" /></a>
                    </li>
                  </ul>
                </div>
                <div className="footer-col col-md-4">
                  <h3>LeanJS</h3>
                  <p>
                    <a href="https://leanjs.com/" target="_blank" rel="noopener noreferrer">LeanJS </a>
                    <span>
                       is a boutique of excellence focused on Lean, JavaScript, UX, Agile, and <strong>people</strong>.
                    </span>
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div className="footer-below">
            <div className="container">
              <div className="row">
                <div className="col-lg-12">
                  <span>Copyright &copy;</span> <a href="https://leanjs.com/" target="_blank" rel="noopener noreferrer">LeanJS</a>
                </div>
              </div>
            </div>
          </div>
        </footer>
      </div>
    )
  }
}

export default Main
