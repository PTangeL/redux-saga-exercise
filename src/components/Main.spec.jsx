import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Main from './Main'

describe('<Main />', () => {

  const wrapper = shallow(
    <Main />
  )

  it('should render', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot()
    })

  it('Method #toggleMenu() should assign beerImgWrapper.style.minHeight', () => {
    //TODO implement test
  })

})
