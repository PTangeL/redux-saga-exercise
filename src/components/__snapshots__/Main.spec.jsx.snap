// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render 1`] = `
<div
  id="page-wrap"
>
  <Menu
    isOpen={true}
    pageWrapId="page-wrap"
    toggleMenu={[Function]}
  />
  <nav
    className="navbar navbar-default navbar-fixed-top navbar-custom"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="https://leanjs.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          LeanJS
        </a>
      </div>
      <ul
        className="nav navbar-nav pull-right"
      >
        <li>
          <button
            className="btn btn-lg btn-outline"
            onClick={[Function]}
          >
            <i
              className="fa fa-heart"
            />
          </button>
        </li>
      </ul>
    </div>
  </nav>
  <Header />
  <section
    id="beers"
  >
    <div
      className="container"
    >
      <Connect(TinderContainer) />
    </div>
  </section>
  <footer
    className="text-center"
  >
    <div
      className="footer-above"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="footer-col col-md-4"
          >
            <h3>
              Main Location
            </h3>
            <p>
              <span>
                1 Fore Stree
              </span>
              <br />
              <span>
                London, EC2Y 5EJ
              </span>
              <br />
              <span
                className="fa fa-map-marker"
              />
              <a
                href="https://leanjs.com/training/react-redux-graphql-london"
                rel="noopener noreferrer"
                target="_blank"
              >
                 Other Locations 
              </a>
            </p>
          </div>
          <div
            className="footer-col col-md-4"
          >
            <h3>
              Around the Web
            </h3>
            <ul
              className="list-inline"
            >
              <li>
                <a
                  className="btn-social btn-outline"
                  href="https://github.com/leanjscom"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i
                    className="fa fa-fw fa-github"
                  />
                </a>
              </li>
              <li>
                <a
                  className="btn-social btn-outline"
                  href="https://twitter.com/leanjscom"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i
                    className="fa fa-fw fa-twitter"
                  />
                </a>
              </li>
              <li>
                <a
                  className="btn-social btn-outline"
                  href="https://www.instagram.com/leanjscom/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i
                    className="fa fa-fw fa-instagram"
                  />
                </a>
              </li>
            </ul>
          </div>
          <div
            className="footer-col col-md-4"
          >
            <h3>
              About LeanJS
            </h3>
            <p>
              <a
                href="https://leanjs.com/"
                rel="noopener noreferrer"
                target="_blank"
              >
                LeanJS 
              </a>
              <span>
                is a boutique of excellence focused on Lean, JavaScript, UX, Agile, and people.
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer-below"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-12"
          >
            <span>
              Copyright Â©
            </span>
             
            <a
              href="https://leanjs.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              LeanJS
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
